src/connectionManager.ts

```diff
@@ -0,0 +1,84 @@
+// TODO: Reference a library instead for GUID generation?
+export type GUID = string & { isGuid: true };
+export function guid(guid: string): GUID {
+  return guid as GUID;
+}
+
+export function generateGUID(): string {
+  const timestamp = new Date().getTime();
+  const randomNum = Math.floor(Math.random() * 1000000);
+  return `${timestamp}-${randomNum}`;
+}
+
+export interface Context {
+  id: GUID;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: GUID;
+  url: URL;
+  active: boolean;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
+
+  constructor() {
+    this.connections = [];
+  }
+
+  public getAllConnections(): Connection[] {
+    return this.connections;
+  }
+
+  public getCurrentConnection(): Connection | undefined {
```

Comment:
Do you have a preference for `undefined` vs `null` for "no result" cases? I don't feel strongly either way, but we should establish conventions like that as a team now.

Reply:
I think I like undefined. Our codebase currently uses undefined and the typescript docs say to use undefined instead of null. [https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined). I think it makes sense for optional variables, but I'm also new to the language.